File: longest common prefix.py
def longest_common_prefix(strings):
    if not strings:
        return ""
    prefix=strings[0]
    for string in strings[1:]:
        while not string.startswith(prefix):
            prefix=prefix[:-1]
            if not prefix:
                return ""
    return prefix

# Example usage
A1 = ["abcdefgh", "aefghijk", "abcefgh"]
A2 = ["abab", "ab", "abcd"]

print("Output 1:", longest_common_prefix(A1))  # Output: "a"
print("Output 2:", longest_common_prefix(A2))   # Output: "ab"




# Example:

# First Iteration:

# prefix = "abcdefgh"
# string = "aefghijk"
# "aefghijk".startswith("abcdefgh") → False
# prefix = "abcdefg"
# "aefghijk".startswith("abcdefg") → False
# Repeat until prefix = "a".
# "aefghijk".startswith("a") → True.
# Second Iteration:

# prefix = "a"
# string = "abcefgh"
# "abcefgh".startswith("a") → True

File: LongestCommonPrefix.java
public class LongestCommonPrefix {
   public static String longestCommonPrefix(String[] strings) {
        if (strings == null || strings.length == 0) {
            return ""; 
        }
     String prefix=strings[0];
     for(int i=1;i<strings.length;i++){
        while(!strings[i].startsWith(prefix)){
            prefix=prefix.substring(0,prefix.length()-1);
            if(prefix.isEmpty()){
                return "";
            }
        }
     }
        
return prefix;

    }

    public static void main(String[] args) {
        String[] A1 = {"abcdefgh", "aefghijk", "abcefgh"};
        String[] A2 = {"abab", "ab", "abcd"};

        System.out.println("Output 1: " + longestCommonPrefix(A1)); // Output: "a"
        System.out.println("Output 2: " + longestCommonPrefix(A2)); // Output: "ab"
    }
}
